packages/trip-details/src/index.tsx

```diff
@@ -0,0 +1,247 @@
+ */
+function DefaultCaloriesDetails({
+  bikeSeconds,
+  calories,
+  walkSeconds
+}: CaloriesDetailsProps): ReactElement {
+  return (
+    <FormattedMessage
+      defaultMessage={defaultMessages["otpUi.TripDetails.caloriesDescription"]}
+      description="Text describing how the calories relate to the walking and biking duration of a trip."
+      id="otpUi.TripDetails.caloriesDescription"
+      values={{
+        b: BoldText,
+        bikeMinutes: Math.round(bikeSeconds / 60),
+        calories: Math.round(calories),
+        dietaryLink,
+        walkMinutes: Math.round(walkSeconds / 60)
+      }}
+    />
+  );
+}
+
+/**
+ * Renders trip details such as departure instructions, fare amount, and calories spent.
+ */
+export function TripDetails({
+  CaloriesDetails = DefaultCaloriesDetails,
+  className = "",
+  DepartureDetails = null,
+  FareDetails = null,
+  itinerary
+}: TripDetailsProps): ReactElement {
+  let companies = "";
+  itinerary.legs.forEach(leg => {
+    if (leg.tncData) {
+      companies = leg.tncData.company;
+    }
+  });
+
+  // process the transit fare
+  const fareResult = coreUtils.itinerary.calculateFares(itinerary);
+  const { currencyCode, maxTNCFare, minTNCFare, transitFare } = fareResult;
+  let fare;
+  if (transitFare || minTNCFare) {
+    fare = (
+      <Styled.Fare>
+        {transitFare && (
+          <Styled.TransitFare>
+            <FormattedMessage
+              defaultMessage={defaultMessages["otpUi.TripDetails.transitFare"]}
+              description="Text showing the price of tickets on public transportation."
+              id="otpUi.TripDetails.transitFare"
+              values={{
+                b: BoldText,
+                transitFare: (
+                  <FormattedNumber
+                    currency={currencyCode}
+                    // For dollars in locales such as 'fr',
+                    // this will limit the display to just the dollar sign
+                    // (otherwise it will render e.g. '2,50 $US' instead of '2,50 $').
+                    currencyDisplay="narrowSymbol"
+                    value={transitFare / 100}
+                    // eslint-disable-next-line react/style-prop-object
+                    style="currency"
+                  />
+                )
+              }}
+            />
+          </Styled.TransitFare>
+        )}
+        {minTNCFare !== 0 && (
+          <Styled.TNCFare>
+            <br />
+            <FormattedMessage
+              defaultMessage={defaultMessages["otpUi.TripDetails.tncFare"]}
+              description="Text showing the price paid to transportation network companies."
+              id="otpUi.TripDetails.tncFare"
+              values={{
+                b: BoldText,
+                companies: (
+                  <Styled.TNCFareCompanies>
+                    {companies.toLowerCase()}
+                  </Styled.TNCFareCompanies>
+                ),
+                maxTNCFare: (
+                  <FormattedNumber
```

Comment:
We seem to be using a few of the same props on a few of these `FormattedNumber`s. Do you think it's worth putting it into a helper method/wrapper component?

Reply:
Good idea, refactored in https://github.com/opentripplanner/otp-ui/pull/281/commits/d39e81032da5df5266a7dfd0e664940c9cdbdd5f.